rtloader/common/builtins/datadog_agent.c

```diff
@@ -488,6 +488,15 @@ static PyObject *set_external_tags(PyObject *self, PyObject *args)
             _free(tags[j]);
         }
         _free(tags);
+
+        if (hostname) {
```

Comment:
This was already freed right after, below the "done" label.

Reply:
This is one of the leaks we had, if the list has multiple tuples we will only free the last hostname and source_type, but not those for the previous entries. It was a small and sneaky leak. Since we break the loop as well, we need to keep both sets of `frees` :)

Reply:
Oh, this free is within the loop! Makes sense now.

Reply:
Would make sense to move this part to a different PR to merge it for 6.14? If I'm not mistaken, `set_external_tags` is quite used, isn't it? That should justify fixing the memory leak in 6.14.

Reply:
It's not a regression so we can skip it for `6.14.0`, it would've been nice to have it in there, but unfortunately it's a little too close to the release date.